src/Handler/NotFoundHandler.php

```diff
@@ -59,15 +59,14 @@ public function __construct(
      * Creates and returns a 404 response.
      *
      * @param ServerRequestInterface $request Passed to internal handler
-     * @param RequestHandlerInterface $handler Ignored.
      */
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
-        if (! $this->renderer) {
+        if ($this->renderer === null) {
```

Comment:
why?

Reply:
Not a bug of course, the type is `null|TemplateRendererInterface`. This is just a proposal to include phpstan as q&a tool without adding too much exclusions (ignore patterns in phpstan.neon). 

The same idea applies to the next comments. It's just a proposal feel free to ignore the P/R. 

I'll update with phpstan, so you can see by yourself and decide if it  helps or not.

Reply:
Fine, check my last comment: https://github.com/zendframework/zend-expressive/pull/565#discussion_r170020405

The condition is replaced wrong, so not sure if the tool works correctly?

Reply:
> The same idea applies to the next comments. It's just a proposal feel free to ignore the P/R.

Having a static analysis tool is very valuable, so I'm fairly sure this will go in :+1: